drivers/nrf_radio_802154/platform/clock/nrf_802154_clock_zephyr.c

```diff
@@ -88,7 +88,7 @@ void nrf_802154_clock_hfclk_stop(void)
 
     hfclk_is_running = false;
```

Comment:
I think this instruction should be executed after `onoff_cancel_or_release()`. If `hfclk_on_callback()` is triggered from IRQ during `onoff_cancel_or_release()` it sets `hfclk_is_running = true;`, after that HFClk is released and flag remains in invalid `true` state.